frontend/src/metabase/query_builder/components/QueryModals.tsx

```diff
@@ -101,6 +101,12 @@ class QueryModals extends Component<QueryModalsProps> {
       setQueryBuilderMode,
     } = this.props;
 
+    // for models, we need to use the card's original dataset_query, not the model's dataset_query
+    // since it only refers to a source table rather than anything else
+    const questionCopy = question.isDataset()
+      ? new Question(this.props.card)
+      : question;
```

Comment:
The existence of this comment generates questions.

Is there a reason why we don't want to use card's original `dataset_query` for non-models?

Like this:

```suggestion
    const questionCopy = new Question(this.props.card);
```

Reply:
It's something I've been wrestling with as well. One of the reasons I didn't want to make a wholesale change is simply because there's so many branches in `QueryModals`, and I'm not 100% confident that using this question copy for everything wouldn't break something.

Otherwise, we could copy the question in the constructor and use that as the `question` object instead of `this.props.question`. I'm not opposed to it but maybe I'm just being a little paranoid about breaking something 😅

Reply:
judging from
https://github.com/metabase/metabase/blob/b92caa60712a7b413c2385075b824700e583efce/frontend/src/metabase/query_builder/selectors.js#L331-L340

It's where we replace `dataset_query` of models which causes this bug. I think if we just want to use question's card you don't have to construct a new question at all.

If you look at `Question`'s `card()` getter, it just returns `_card` which is passed from the constructor in the first place.
https://github.com/metabase/metabase/blob/b92caa60712a7b413c2385075b824700e583efce/frontend/src/metabase-lib/Question.ts#L146-L152

I'd just leave everything the same but spread `this.props.card` instead.